service/orchestrator/assessment_results.go

```diff
@@ -40,9 +40,35 @@ import (
 	"clouditor.io/clouditor/api/assessment"
 	"clouditor.io/clouditor/api/orchestrator"
 	"clouditor.io/clouditor/internal/logging"
+	"clouditor.io/clouditor/persistence"
 	"clouditor.io/clouditor/service"
 )
 
+// GetAssessmentResult gets one assessment result by id
+func (svc *Service) GetAssessmentResult(ctx context.Context, req *orchestrator.GetAssessmentResultRequest) (res *assessment.AssessmentResult, err error) {
+
+	// Validate request
+	if service.ValidateRequest(req) != nil {
+		return
+	}
+
+	// Check, if this request has access to the cloud service according to our authorization strategy.
+	if !svc.authz.CheckAccess(ctx, service.AccessRead, req) {
```

Comment:
This will probably not work, because the request does not contain the cloud service ID, only the ID for a specific assessment result. 

We probably need to add an additional query, to check if `cloud_service_id` of the result is within `allowed`. There should be similar code already for list-assessment results.

Reply:
Alternatively, we restrict the quest is only retrieving a result based on id + cloud service id. But then we still need to update the query in line 62 and the HTTP API annotations in the protocol file.

Reply:
GetAssessmentResultRequest only has one field -> id. 
GetAssessmentResult() checks if cloud_service_id is allowed.
Test cases are updated and expanded appropriately.